File: 프로그래머스/2/12953. N개의 최소공배수/N개의 최소공배수.java
class Solution {
    public int solution(int[] arr) {
    
        int answer = lcm(arr[0], arr[1]); // 최소 공배수 => 곱 / 최대 공약수
        
        for(int i=2; i<arr.length; i++){
            answer = lcm(answer, arr[i]);
        }
        
        return answer;
    }
    
    public int gcd(int a, int b){
        if (b == 0) return a;
        return gcd(b, a % b);
    }
    
    public int lcm(int a, int b){
        return (a * b) / gcd(a, b);
    }
}

File: 프로그래머스/lv2/12953. N개의 최소공배수/N개의 최소공배수.py

from math import gcd # 최대공약수 

def solution(arr):
    for i in range(1, len(arr)):
        arr[i] = arr[i-1] * arr[i] // gcd(arr[i-1], arr[i]) # 최소공배수 = 두 수 // 최소공배수
    return arr[-1]